herd/valconstraint.ml

```diff
@@ -32,6 +32,8 @@ module type S = sig
 
   type expr =
     | Atom of atom
+    | Address of atom
+    | Deref of atom
```

Comment:
I am wondering: would'nt be possible to have  the address and deref operations implemented as unary operation. That is extend type op1 in lib/op.{mi,mli} with constructors Address and Deref, then encode your `Address at` by `Unop (Address,at)`.
This would simplify some functions a bit.

Reply:
it's difficult to say, on the one hand these are special unary operations in eventsMonad, and we add invariants corresponding to the address of each location`&x = 0xf00`. However these operations should be used only in data access instructions like ldr etc... and require the inclusion of a symbol table (`symbs` in mem.ml) so they are not normal.

They act like normal unary operations, but they need special treatment with the symbol table. If we were to treat them as unary operations,  then we would need to pass the symbol table into op/symbValue etc... I think it's simpler, and more elegant, to restrict the scope of the symbol table to the initial state in litmus tests, and the frontend of herd where such constraints are parsed and sent to the solver only. 

This would prevent numeric addresses from leaking into the symbolic part of herd as well which would prevent accidental conflation of symbols/addresses in the rest of herd.